src/aggregator/aggregator/generic_elem.go

```diff
@@ -742,12 +742,13 @@ func (e *GenericElem) findOrCreate(
 			sourcesSeen = make(map[uint32]*bitset.BitSet)
 		}
 	}
+	// lockedAggregation will be returned to pool on timedAggregation close
 	timedAgg = timedAggregation{
 		startAt: alignedStart,
-		lockedAgg: &lockedAggregation{
-			sourcesSeen: sourcesSeen,
-			aggregation: e.NewAggregation(e.opts, e.aggOpts),
-		},
+		lockedAgg: lockedAggregationFromPool(
```

Comment:
Super nit: maybe add a comment linking to this PR just to have an easy tie back to the reasoning (outside of git blame, of course).